src/tasks/netSdk/NetSdkTaskHelper.ts

```diff
@@ -0,0 +1,138 @@
+            withNamedArg('--arch', await this.normalizeArchitectureToRid()),
+            withArg(publishFlag),
+            withNamedArg('--configuration', configuration),
+            withNamedArg('-p:ContainerImageName', getValidImageName(folderName.name), { assignValue: true }),
+            withNamedArg('-p:ContainerImageTag', IMAGE_TAG, { assignValue: true })
+        )();
+
+        const quotedArgs = Shell.getShellOrDefault().quote(args);
+        return quotedArgs.join(' ');
+    }
+
+    public async getNetSdkRunCommand(context: DockerTaskExecutionContext): Promise<string> {
+        const client = await ext.runtimeManager.getClient();
+        const folderName = await quickPickWorkspaceFolder(context.actionContext, ERR_MSG_NO_WORKSPACE_FOLDER);
+
+        const command = await client.runContainer({
+            detached: true,
+            publishAllPorts: true,
+            name: getDefaultContainerName(folderName.name, IMAGE_TAG),
+            environmentVariables: {},
+            removeOnExit: true,
+            imageRef: getDefaultImageName(folderName.name, IMAGE_TAG),
+            labels: defaultVsCodeLabels,
+            mounts: await this.getRemoteDebuggerMount(),
+            customOptions: '--expose 8080',
+            entrypoint: '/bin/sh'
+        });
+
+        const quotedArgs = Shell.getShellOrDefault().quote(command.args);
+        const commandLine = [client.commandName, ...quotedArgs].join(' ');
+
+        return commandLine;
+    }
+
+    public async inferProjPath(context: IActionContext, folder: WorkspaceFolder): Promise<string> {
+        const noProjectFileErrMessage = l10n.t('No .csproj file could be found.');
+        const item = await quickPickProjectFileItem(context, undefined, folder, noProjectFileErrMessage);
+        return item.absoluteFilePath || '';
+    }
+
+    /**
+     * This method normalizes the Docker OS type to match the .NET Core SDK conventions.
+     * {@link https://learn.microsoft.com/en-us/dotnet/core/rid-catalog}
+     */
+    private async normalizeOsToRid(): Promise<'linux' | 'win'> {
+        const dockerOsType = await getDockerOSType();
+        return dockerOsType === 'windows' ? 'win' : 'linux';
+    }
+
+    /**
+     * This method normalizes the native architecture to match the .NET Core SDK conventions.
+     * {@link https://learn.microsoft.com/en-us/dotnet/core/rid-catalog}
+     */
+    private async normalizeArchitectureToRid(): Promise<RidCpuArchitecture> {
+        const architecture = os.arch();
+        switch (architecture) {
+            case 'x32':
+            case 'ia32':
+                return 'x86';
+            default:
+                return architecture;
+        }
+    }
+
+    /**
+     * This methods returns the mount for the remote debugger ONLY as the SDK built container will have
+     * everything it needs to run the app already inside.
+     */
+    private async getRemoteDebuggerMount(): Promise<RunContainerBindMount[] | undefined> {
```

Comment:
Consider adding a shared utility function that can be re-used both here and in NetCoreTaskHelper.ts which takes an OSType and returns the appropriate debugger, since the logic for setting a debugger path should be the same between the two scenarios.

Reply:
I like the idea but after playing around with it, I think this might make the code in `NetCoreTaskHelper.ts` more confusing. We could add a method to `TaskHelper` called  `addDebuggerVolumeWithoutConflict()` and it would make `inferVolumes()` look like this. Which IMO is more confusing than what we have now. 

``` TypeScript
...
addVolumeWithoutConflicts(volumes, appVolume);
addVolumeWithoutConflicts(volumes, srcVolume);
addDebuggerVolumeWithoutConflicts(volumes, osType);
if (await fse.pathExists(nugetRootVolume.localPath)) {
    addVolumeWithoutConflicts(volumes, nugetRootVolume);
}
...
```